tests/cephfs/cephfs_bugs/test_validate_deadlock_bw_unlink_and_rename.py

```diff
@@ -0,0 +1,153 @@
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(10))
+        )
+        log.info("Mount 1 subvolume on kernel and 1 subvolume on Fuse → Client1")
+
+        kernel_mounting_dir_1 = f"/mnt/cephfs_kernel{mounting_dir}_1/"
+        mon_node_ips = fs_util_v1.get_mon_node_ips()
+        subvol_path_kernel, rc = clients[0].exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume getpath {default_fs} {subvolume_name}_1 {subvolume_group_name}",
+        )
+        fs_util_v1.kernel_mount(
+            [clients[0]],
+            kernel_mounting_dir_1,
+            ",".join(mon_node_ips),
+            sub_dir=f"{subvol_path_kernel.strip()}",
+            extra_params=f",fs={default_fs}",
+        )
+
+        fuse_mounting_dir_1 = f"/mnt/cephfs_fuse{mounting_dir}_1/"
+        subvol_path_fuse, rc = clients[0].exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume getpath {default_fs} {subvolume_name}_2 {subvolume_group_name}",
+        )
+        fs_util_v1.fuse_mount(
+            [clients[0]],
+            fuse_mounting_dir_1,
+            extra_params=f" -r {subvol_path_fuse.strip()} --client_fs {default_fs}",
+        )
+
+        log.info("Run the unlink and reintegrate straydn(rename) for around 30mins")
+        kernel_file_path = f"{kernel_mounting_dir_1}primary_file_k"
+        fuse_file_path = f"{fuse_mounting_dir_1}primary_file_f"
+        run_time = 30 * 60  # 30 minutes * 60 seconds/minute
+        start_time = time.time()
+        # Step 1: Create a file and hard links to 5 different files
+        while time.time() - start_time < run_time:
+            clients[0].exec_command(sudo=True, cmd=f"touch {kernel_file_path}")
+            clients[0].exec_command(sudo=True, cmd=f"touch {fuse_file_path}")
+            for i in range(1, 6):
+                clients[0].exec_command(
+                    sudo=True,
+                    cmd=f"ln {kernel_file_path} {kernel_mounting_dir_1}hardlink{i}_file_k",
+                )
+                clients[0].exec_command(
+                    sudo=True,
+                    cmd=f"ln {fuse_file_path} {fuse_mounting_dir_1}hardlink{i}_file_f",
+                )
+
+            # Step 2: Remove the file and hard links
+            for i in range(6):
+                clients[0].exec_command(
+                    sudo=True,
+                    cmd=f"rm -rf {kernel_mounting_dir_1}{'primary_file_k' if i == 0 else f'hardlink{i}_file_k'}",
+                )
+                clients[0].exec_command(
+                    sudo=True,
+                    cmd=f"rm -rf {fuse_mounting_dir_1}{'primary_file_f' if i == 0 else f'hardlink{i}_file_f'}",
+                )
+        time.sleep(1)
+
+        log.info(
+            "Check for the Ceph Health to see if there are any deadlock bw unlink and rename."
+        )
+        ceph_health = fs_util_v1.get_ceph_health_status(clients[0])
+        log.info(ceph_health)
```

Comment:
we are not taking an action(raise exception) if ceph health is bad, is that ok?
Also, we are printing 'None' here, can we print actual health status into test log?

2023-09-12 01:31:27,132 (cephci.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename) [INFO] - cephci.tests.cephfs.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename.py:117 - Check for the Ceph Health to see if there are any deadlock bw unlink and rename.
2023-09-12 01:31:27,133 (cephci.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename) [INFO] - cephci.ceph.ceph.py:1558 - Running command ceph -s -f json on 10.0.211.12 timeout 600
2023-09-12 01:31:27,486 (cephci.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename) [INFO] - cephci.ceph.ceph.py:1592 - Command completed successfully
2023-09-12 01:31:27,487 (cephci.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename) [INFO] - cephci.tests.cephfs.cephfs_bugs.test_validate_deadlock_bw_unlink_and_rename.py:119 - None

Reply:
Updated the script to raise exceptions.